File: adapter/exercise/JsonReportAdaptee.java
package adapter.exercise;

public class JsonReportAdaptee implements ReportingServiceAdapter {
    private final ThirdPartyReportingService thirdPartyReportingService;

    public JsonReportAdaptee(ThirdPartyReportingService thirdPartyReportingService) {
        this.thirdPartyReportingService = thirdPartyReportingService;
    }

    @Override
    public JSONReport generateReport() {
        return convertDataIntoJson(thirdPartyReportingService.generateReport());
    }

    private JSONReport convertDataIntoJson(CSVReport csvReport) {
        System.out.println("Converting data into json..");
        return new JSONReport();
    }
}

File: typescript/adapter/exercises/json-report-adaptee.ts
import { JSONReport } from "./json-report";
import { CSVReport } from "./cv-report";
import { ReportingServiceAdapter } from "./reporting-service-adapter";
import { ThirdPartyReportingService } from "./third-party-reporting-service";

class JsonReportAdaptee implements ReportingServiceAdapter {
  private readonly thirdPartyReportingService: ThirdPartyReportingService;

  constructor(thirdPartyReportingService: ThirdPartyReportingService) {
    this.thirdPartyReportingService = thirdPartyReportingService;
  }

  generateReport(): JSONReport {
    return this.convertDataIntoJson(
      this.thirdPartyReportingService.generateReport()
    );
  }

  private convertDataIntoJson(csvReport: CSVReport): JSONReport {
    console.log("Converting data into JSON...");
    return new JSONReport();
  }
}
